### PR TITLE
Update crun to 1.17

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Update crun to the latest release.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by ccojocar — APPROVED** [review]

**Comment by mhils** [comment]
CI fails with `cat: /proc//attr/current: No such file or directory`, which comes from here:

https://github.com/kubernetes-sigs/security-profiles-operator/blob/474af83f88d8950548c5dcc54352e31f6504e87b/hack/ci/e2e-apparmor.sh#L85-L89

No idea why this is failing though. 🙈

**Comment by saschagrunert** [comment]
@mhils the container creation fails with:
```
Container creation error: write file `/proc/thread-self/attr/apparmor/exec`: No such file or directory
```

**Comment by saschagrunert** [comment]
We need to dig deeper into the AppArmor issue. Let's get this in to unblock the CI.

@kubernetes-sigs/security-profiles-operator-maintainers PTAL

**Review by pjbgf — APPROVED** [review]

**Pull request merged**
